src/aosm/azext_aosm/configuration_models/onboarding_cnf_input_config.py

```diff
@@ -94,13 +62,17 @@ def validate(self):
 class OnboardingCNFInputConfig(OnboardingNFDBaseInputConfig):
     """Input configuration for onboarding CNFs."""
 
-    # TODO: Add better comment for images as not a list
-    images: ImageSourceConfig = field(
-        default_factory=ImageSourceConfig,
+    image_sources: list = field(
+        default_factory=lambda: [],

```

Comment:
Perhaps this is clearer:

```suggestion
        default_factory=list,

```